README.md

```diff
@@ -1,8 +1,62 @@
-This plugin does a HTTP/HTTPS request to a speficied URL with the build parameters automatically
-part of the payload as key/value pairs.
+# Http Request Plugin for Jenkins
 
-It can be configured whether GET or POST (default) should be used as HTTP method, as well
-as if the status return code of the response marks the build as failed (default) or is ignored.
-You can specify different username / password pairs in the global configuration to support
-authenticated HTTP/HTTPS requests, which you then refer to from your build configuration.
+This plugin sends a HTTP/HTTPS request to a user speficied URL.
 
+## Features
+
+The following features are available in both Pipeline and traditional
+project types:
+
+* Programmable HTTP method: GET, POST, PUT, DELETE, or HEAD
+* Programmable range of expected response codes (a response code outside the range fails the build)
+* Supports Basic Authentication (see global configuration)
+* Supports From Authentication (see global configuration)
+* You can specify a string that must be present in the response (if the string is not present, the build fails)
+* You can set a connection timeout limit (build fails if timeout is exceeded)
+* You can set an "Accept" header
+* You can set a "Content-type" header
+
+### Basic plugin features
+
+The following features are only present in the non-pipeline version of
+the plugin. For the Pipeline version, these features are available
+programmatically.
+
+* You can send the build parameters as URL query strings
+* You can store the response to a file, built-in to the plugin
+
+### Pipeline features
+
+In a Pipeline job, you have total control over how the url is
+formed. Suppose you have a build parameter called "param1",
+you can pass it to the HTTP request programmatically like so:
+
+```
+httpRequest url:"http://httpbin.org/response-headers?param1=${param1}"
```

Comment:
BTW the `url:` is optional when only a single parameter is being given (that of the `@DataBoundConstructor`). So this would also work:

```groovy
httpRequest "http://httpbin.org/response-headers?param1=${param1}"
```